lib/IMAP/MessageMapper.php

```diff
@@ -317,17 +317,20 @@ public function findByIds(Horde_Imap_Client_Base $client,
 	 * @param string $sourceFolderId
 	 * @param int $messageId
 	 * @param string $destFolderId
+	 * @return int the new UID
 	 */
 	public function move(Horde_Imap_Client_Base $client,
 		string $sourceFolderId,
 		int $messageId,
-		string $destFolderId): void {
+		string $destFolderId): int {
 		try {
-			$client->copy($sourceFolderId, $destFolderId,
+			$mapping = $client->copy($sourceFolderId, $destFolderId,
 				[
 					'ids' => new Horde_Imap_Client_Ids($messageId),
 					'move' => true,
```

Comment:
```suggestion
					'move' => true,
					'force_map' => true,
```

Nice :+1: 


If we get the UID mapping depends on the UIDPLUS extension. 
Copy returns either an array (with UIDPLUS) or true (without UIDPLUS).

The force_map option tells horde to always return an array. Without the UIDPLUS extension, it tries to look up the UID by the message ID. Not perfect but our best option I guess ;)